File: TeamCode/src/main/java/expo/util/ButtonToggle.java
package expo.util;

public class ButtonToggle {
    private Button button;
    private ExpoGamepad gamepad;

    private boolean toggle;
    private boolean current;
    private boolean last;

    public ButtonToggle(ExpoGamepad gamepad, Button button, boolean toggle) {
        this.gamepad = gamepad;
        this.button = button;
        this.toggle = toggle;
    }

    public ButtonToggle(ExpoGamepad gamepad, Button button) {
        this(gamepad, button, false);
    }


    public void update() {
        last = current;
        current = gamepad.getButton(button);

        //if the button is pressed
        //and it wasnt pressed before
        //then change the toggle
        if (current && !last) {
            toggle = !toggle;
        }
    }

    public boolean getValue() {
        return toggle;
    }
}

File: TeamCode/src/main/java/expo/gamepad/ButtonToggle.kt
package expo.gamepad

import expo.util.ExpoGamepad

class ButtonToggle(private val gamepad: ExpoGamepad, private val button: Button, private var toggle: Boolean) {
    private var current: Boolean = false
    private var last: Boolean = false

    constructor(gamepad: ExpoGamepad, button: Button) : this(gamepad, button, false)

    fun update() {
        last = current
        current = if (button == Button.A || button == Button.B)
            gamepad.getButton(button) && !gamepad.getButton(Button.START)
        else
            gamepad.getButton(button)
        //if the button is pressed
        //and it wasnt pressed before
        //then change the toggle

        if (!last && current) toggle = !toggle
    }

    fun getValue() = toggle
}